src/Microsoft.Azure.WebJobs.Host/Queues/QueueProcessor.cs

```diff
@@ -119,6 +127,9 @@ public QueueProcessor(QueueProcessorFactoryContext context)
                 if (message.DequeueCount >= MaxDequeueCount)
                 {
                     await CopyMessageToPoisonQueueAsync(message, _poisonQueue, cancellationToken);
+
+                    message.SetProperties(id, popReceipt, insertionTime, nextVisibleTime, expirationTime);
```

Comment:
Can we put a simple conditional around this so we only do these updates if the pop receipt (and/or other values) have changed? That would keep this scoped to only scenarios where people are using 8.x, minimizing the chances of breaking changes or odd behaviors. Another possibility could be to make it conditional on the Azure SDK version.